core/src/main/java/org/elasticsearch/index/shard/TranslogRecoveryPerformer.java

```diff
@@ -157,7 +158,8 @@ private void performRecoveryOperation(Engine engine, Translog.Operation operatio
                     Translog.Index index = (Translog.Index) operation;
                     // we set canHaveDuplicates to true all the time such that we de-optimze the translog case and ensure that all
                     // autoGeneratedID docs that are coming from the primary are updated correctly.
-                    Engine.Index engineIndex = IndexShard.prepareIndex(docMapper(index.type()), source(shardId.getIndexName(), index.type(), index.id(), index.source())
+                    Engine.Index engineIndex = IndexShard.prepareIndex(docMapper(index.type()),
+                        source(shardId.getIndexName(), index.type(), index.id(), index.source(), XContentFactory.xContentType(index.source()))
```

Comment:
Yeah there is still auto-detection here; I can modify this but don't have a good sense of knowing if we should do this in a minor version or not